astronomer/providers/google/cloud/triggers/dataproc.py

```diff
@@ -87,6 +88,78 @@ async def run(self) -> AsyncIterator["TriggerEvent"]:  # type: ignore[override]
         yield TriggerEvent({"status": "error", "message": "Timeout"})
 
 
+class DataprocDeleteClusterTrigger(BaseTrigger, ABC):
+    """
+    Asynchronously check the status of a cluster
+
+    :param cluster_name: The name of the cluster
+    :param end_time: Time in second left to check the cluster status
+    :param project_id: The ID of the Google Cloud project the cluster belongs to
+    :param region: The Cloud Dataproc region in which to handle the request
+    :param metadata: Additional metadata that is provided to the method
+    :param gcp_conn_id: The connection ID to use when fetching connection info.
+    :param polling_interval: Time in seconds to sleep between checks of cluster status
+    """
+
+    def __init__(
+        self,
+        cluster_name: str,
+        end_time: float,
+        project_id: Optional[str] = None,
+        region: Optional[str] = None,
+        metadata: Sequence[Tuple[str, str]] = (),
+        gcp_conn_id: str = "google_cloud_default",
+        polling_interval: float = 5.0,
+        **kwargs: Any,
+    ):
+        super().__init__(**kwargs)
+        self.cluster_name = cluster_name
+        self.end_time = end_time
+        self.project_id = project_id
+        self.region = region
+        self.metadata = metadata
+        self.gcp_conn_id = gcp_conn_id
+        self.polling_interval = polling_interval
+
+    def serialize(self) -> Tuple[str, Dict[str, Any]]:
+        """Serializes DataprocDeleteClusterTrigger arguments and classpath."""
+        return (
+            "astronomer.providers.google.cloud.triggers.dataproc.DataprocDeleteClusterTrigger",
+            {
+                "cluster_name": self.cluster_name,
+                "end_time": self.end_time,
+                "project_id": self.project_id,
+                "region": self.region,
+                "metadata": self.metadata,
+                "gcp_conn_id": self.gcp_conn_id,
+                "polling_interval": self.polling_interval,
+            },
+        )
+
+    async def run(self) -> AsyncIterator["TriggerEvent"]:  # type: ignore[override]
+        """Wait until cluster is deleted completely"""
+        hook = DataprocAsyncHook(gcp_conn_id=self.gcp_conn_id)
+        while self.end_time > time.monotonic():
+            try:
+                cluster = await hook.get_cluster(
+                    region=self.region,  # type: ignore[arg-type]
+                    cluster_name=self.cluster_name,
+                    project_id=self.project_id,  # type: ignore[arg-type]
+                    metadata=self.metadata,
+                )
+                self.log.info(
+                    "Cluster status is %s. Sleeping for %s seconds.",
+                    cluster.status.state,
+                    self.polling_interval,
+                )
+                await asyncio.sleep(self.polling_interval)
+            except NotFound:
+                yield TriggerEvent({"status": "success", "message": ""})
```

Comment:
why arent we sending message something like "cluster with id <cluster id> deleted successfully" here ?

Reply:
sync version just log `Cluster deleted` in case of success